### PR TITLE
[AAE-3321] Select uploaded local files by default

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/AAE-3321

**What is the new behaviour?**

content-node-panel-preselect

document-list-preselect

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/document-list/components/document-list.component.ts [hunk 1]
```diff
@@ -257,7 +256,40 @@ export class DocumentListComponent implements OnInit, OnChanges, OnDestroy, Afte
...
-    currentFolderId: string = null;
+    // currentFolderId: string = null;
+    _currentFolderId: string = null;
+
+    @Input()
+    preSelectedNodes: any[] = [];
+
+    /** The ID of the folder node to display or a reserved string alias for special sources */
+    @Input()
+    set currentFolderId(currentFolderId: string) {
```
Comment (mauriziovitale) [inline lib/content-services/src/lib/document-list/components/document-list.component.ts:267]: Why are we putting this method back ?

File: lib/content-services/src/lib/document-list/components/document-list.component.ts [hunk 2]
```diff
@@ -469,7 +488,13 @@ export class DocumentListComponent implements OnInit, OnChanges, OnDestroy, Afte
 
         if (changes['currentFolderId']?.currentValue !== changes['currentFolderId']?.previousValue) {
             if (this.data) {
-                this.data.loadPage(null, false);
+                if (this.hasPreSelectedNodes()) {
+                    this.data.loadPage(null, false, false, this.preSelectedNodes);
+                    this.onNodeSelect({ row: <ShareDataRow> this.data.getPreSelectedRows()[0], selection: <ShareDataRow[]> this.data.getPreSelectedRows() });
```
Comment (mauriziovitale) [inline lib/content-services/src/lib/document-list/components/document-list.component.ts:493]: what are we trying to do here?

File: lib/content-services/src/lib/document-list/components/document-list.component.ts [hunk 3]
```diff
@@ -495,7 +525,12 @@ export class DocumentListComponent implements OnInit, OnChanges, OnDestroy, Afte
         this.ngZone.run(() => {
             this.resetSelection();
             if (this.node) {
-                this.data.loadPage(this.node, this._pagination.merge);
+                if (this.hasPreSelectedNodes()) {
+                    this.data.loadPage(this.node, this._pagination.merge, null, this.preSelectedNodes);
+                    this.onNodeSelect({ row: <ShareDataRow> this.data.getPreSelectedRows()[0], selection: <ShareDataRow[]> this.data.getPreSelectedRows() });
```
Comment (mauriziovitale) [inline lib/content-services/src/lib/document-list/components/document-list.component.ts:530]: if the intension is to bubble up the selection, I don't think we need it ...

File: lib/content-services/src/lib/document-list/components/document-list.component.ts (hunk 1 above)
Comment (sivakumar414ram) [reply lib/content-services/src/lib/document-list/components/document-list.component.ts]: Was not rebased. It fixed after rebase

File: lib/content-services/src/lib/document-list/components/document-list.component.ts (hunk 3 above)
Comment (sivakumar414ram) [reply lib/content-services/src/lib/document-list/components/document-list.component.ts]: We need to bubble up only preselectedNodes selection

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts [hunk 4]
```diff
@@ -267,6 +271,22 @@ export class ContentNodeSelectorPanelComponent implements OnInit, OnDestroy {
...
+
+        this.uploadService.fileUploadComplete
+        .pipe(
+            debounceTime(300),
+            scan((files, currentFile) => [...files, currentFile], []),
+            takeUntil(this.onDestroy$)
+        )
+        .subscribe((uploadedFiles: FileUploadCompleteEvent[]) => {
+            this.preSelectedNodes = [...uploadedFiles.map((uploadedFile) => uploadedFile.data)];
+            this.onFileUploadEvent();
```
Comment (mauriziovitale) [inline lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts:283]: just use this.documentList.reload()

**Review by mauriziovitale — CHANGES_REQUESTED** [review]
Please add unit test

File: lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts (hunk 4 above)
Comment (sivakumar414ram) [reply lib/content-services/src/lib/content-node-selector/content-node-selector-panel.component.ts]: Updated

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**